### DISCUSSION TITLE
Working accounts with address books

### DISCUSSION THREAD

**Discussion by simeononsecurity** [rustdesk/rustdesk#1132] [General]
I see in the GUI of the client that there is options for an address book and that there is a logon option. However, there is no documentation on how to sign up or where to sign up or how to self host this option. Having this, in addition to the self host option, is the only thing setting you apart from anydesk and one of the biggest things keeping you from the levels of TeamViewer. 
I don't think it's fair to call yourself a TeamViewer alternative if you don't have this feature.

**Comment by tombatti** [comment]
Hi all, any news on this topic? 
I am also very interested in this account management. 
Thank you

**Reply by tovoro** [reply]
Same here, im very interested in this. Any news here? In January somebody from Rustdeck just posted "soon".

**Reply by lbdhr** [reply]
It seems like we cannot get this function in free mode.

**Reply by taos15** [reply]
Is there a reason why this can not be implemented in selfhosting? Is this something the dev won't merge of a PR was open? 

I understand that the cirrent implementation might be using cloud server resources, but will a option for selfhosting this functionality be rejected? Something like a toggle to change between pro version address book, and selfthosted address book? 

Thanks

**Comment by admiraldaro** [comment]
Full disclosure: I’m the author of a small external project that provides a self-hosted option for the account login and address-book functionality discussed here:

https://github.com/admiraldaro/rust-book

Rust-Book is an unofficial account and legacy address-book API server. It is not a patch or pull request for the RustDesk client, and it does not unlock Server Pro features. Instead, it uses the existing **API Server** setting and the legacy address-book protocol already supported by the regular unmodified RustDesk client.

The basic setup is:

1. Run Rust-Book alongside the standard OSS `hbbs` and `hbbr` services.
2. Create an account on your own server using the included CLI tool.
3. Set the RustDesk **API Server** field to the base HTTPS URL of the Rust-Book installation.
4. Log in from the RustDesk client using the account you created.

The client can then synchronize one legacy address book associated with that account. Rust-Book uses PHP and SQLite and also includes a lightweight browser-based administration panel.

Its scope is deliberately limited: it does not provide managed devices, ACLs, centralized settings or the newer multi-address-book API from Server Pro. The current release is also still early and has so far been verified with the official Windows client 1.4.9, so compatibility with other client versions and platforms may vary.

This does not answer whether a native self-hosted implementation would be accepted upstream, but it provides a working external option for people who want to keep the official OSS server and unmodified client while self-hosting the login and legacy address book.
